shared/lib/names.js

```diff
@@ -75,6 +75,26 @@ function packageEntryToPath (packageEntry, { modulePath } = {}) {
   return modulePath !== undefined ? join(dirname(packagePath), modulePath) : packagePath
 }
 
+/**
+ * Resolve path to package field from any npm workspace
+ *
+ * @param {string} packageName - Installed npm package name
+ * @param {PackageOptions} [options] - Package resolution options
+ * @returns {string} Path to installed npm package field
+ */
+function packageFieldToPath (packageName, { modulePath, field = 'main' } = {}) {
```

Comment:
Makes sense, but can you suggest a better name than `resolve()`?

I've used `packageEntryToPath()` for packages that support `require.resolve()`
But used `packageFieldToPath()` for packages that aren't built yet or otherwise unresolvable

For example `packageEntryToPath('govuk-frontend/src/govuk/all.mjs')` crashes due to this bug:

* https://github.com/alphagov/govuk-frontend/issues/3755

So we have to use `packageFieldToPath()` instead

Reply:
How about `packageModulePath` to make explicit that it finds the path to a specific module in the package?

Reply:
Will that be confusing as you could pick the `module`, `main` or `sass` fields?

Doesn't actually have to be a module, just a package field

Reply:
I'm going to use `type` instead of `field`

Where the [**package.json** `"type"` field](https://nodejs.org/api/packages.html#type) can be `module` or `commonjs`

Reply:
How's this? Just pushed up


```mjs
import { paths } from 'govuk-frontend-config'
import { packageTypeToPath } from 'govuk-frontend-lib/names'

// Locate GOV.UK Frontend
const packagePath = packageTypeToPath('govuk-frontend', {
  moduleRoot: paths.stats,
  type: 'module'
})
```